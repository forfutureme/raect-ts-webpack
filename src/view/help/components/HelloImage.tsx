/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-09 08:56:18
 * @Desc: Hello 图片资源引入
 */
import React from 'react'

import Logo from '@image/logo.png'
import Avatar from '@image/avatar.jpg'
import '@svg'

const HelloImage: React.FC = () => {
  return (
    <div>
      <h2>HelloImage</h2>
      <div>
        <h3>Png</h3>
        <img width="100" src={Logo} />
      </div>
      <div>
        <h3>jpg</h3>
        <img width="100" src={Avatar} />
      </div>
      <div>
        <h3>svg</h3>
        <svg>
          <use xlinkHref="#qq"></use>
        </svg>
      </div>
    </div>
  )
}

export default HelloImage
