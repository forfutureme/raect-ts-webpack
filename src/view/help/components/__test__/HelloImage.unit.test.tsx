/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-09 09:27:05
 * @Desc: HelloImage 单元测试
 */
import renderer from 'react-test-renderer'
import React from 'react'

import HelloImage from '../HelloImage'

describe('HelloImage测试', () => {
  it('测试渲染', () => {
    const json = renderer.create(<HelloImage />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
