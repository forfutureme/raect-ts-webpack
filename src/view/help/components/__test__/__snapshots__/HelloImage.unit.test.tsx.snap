// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloImage测试 测试渲染 1`] = `
<div>
  <h2>
    HelloImage
  </h2>
  <div>
    <h3>
      Png
    </h3>
    <img
      src="assets-resources-file"
      width="100"
    />
  </div>
  <div>
    <h3>
      jpg
    </h3>
    <img
      src="assets-resources-file"
      width="100"
    />
  </div>
  <div>
    <h3>
      svg
    </h3>
    <svg>
      <use
        xlinkHref="#qq"
      />
    </svg>
  </div>
</div>
`;
